mysql/mysqldump.go

```diff
@@ -0,0 +1,578 @@
+	var l [][]byte
+	for {
+		b := r.ReadLine()
+		l = append(l, b)
+		// If we see a semicolon or eof, we're likely to have a command, so try to parse it.
+		// Note: we could just parse every iteration, but that would mean more attempts at parsing.
+		if strings.Contains(string(b), ";") || r.EOF {
+			n := 0
+			for i := range l {
+				n += len(l[i])
+			}
+			s := make([]byte, n)
+			n = 0
+			for i := range l {
+				n += copy(s[n:], l[i])
+			}
+			query := string(s)
+			tree, _, err := parser.New().Parse(query, "", "")
+			if err == nil {
+				return s, tree, nil
+			}
+			newTree, err1 := handleParseError(conv, query, err, l)
+			if err1 == false {
+				return s, newTree, nil
+			}
+			// Likely causes of failing to parse:
+			// a) complex statements with embedded semicolons e.g. 'CREATE FUNCTION'
+			// b) a semicolon embedded in a multi-line comment, or
+			// c) a semicolon embedded a string constant or column/table name.
+			// We deal with this case by reading another line and trying again.
+			conv.Stats.Reparsed++
+		}
+		if r.EOF {
+			return nil, nil, fmt.Errorf("Error parsing last %d line(s) of input", len(l))
+		}
+	}
+}
+
+// processStatement extracts schema information from MYSQL
+// statements, updating Conv with new schema information, and returning
+// true if INSERT statement is encountered.
+func processStatement(conv *internal.Conv, stmt ast.StmtNode) bool {
+
+	switch ddlStmt := stmt.(type) {
+	case *ast.CreateTableStmt:
+		if conv.SchemaMode() {
+			processCreateTable(conv, ddlStmt)
+		}
+	case *ast.AlterTableStmt:
+		if conv.SchemaMode() {
+			processAlterTable(conv, ddlStmt)
+		}
+	case *ast.SetStmt:
+		if conv.SchemaMode() {
+			processSetStmt(conv, ddlStmt)
+		}
+	case *ast.InsertStmt:
+		return true
+	default:
+		conv.SkipStatement(reflect.TypeOf(stmt).String())
+	}
+	return false
+}
+
+func processSetStmt(conv *internal.Conv, stmt *ast.SetStmt) {
+	if stmt.Variables != nil && len(stmt.Variables) > 0 {
+		for _, variable := range stmt.Variables {
+			if variable.Name == "TIME_ZONE" {
+				value := variable.Value
+				switch val := value.(type) {
+				case *driver.ValueExpr:
+					if val.GetValue() == nil {
+						logStmtError(conv, stmt, fmt.Errorf("found nil value in set statement"))
+						return
+					}
+					conv.TimezoneOffset = fmt.Sprintf("%v", val.GetValue())
+				default:
+					logStmtError(conv, stmt, fmt.Errorf("found %s type value in set statement", reflect.TypeOf(val)))
+					return
+				}
+			}
+		}
+	}
+}
+
+func processCreateTable(conv *internal.Conv, stmt *ast.CreateTableStmt) {
+	var colNames []string
+	colDef := make(map[string]schema.Column)
+	if stmt.Table == nil {
+		logStmtError(conv, stmt, fmt.Errorf("table is nil"))
+		return
+	}
+	tableName, err := getTableName(stmt.Table)
+	if err != nil {
+		logStmtError(conv, stmt, fmt.Errorf("can't get table name: %w", err))
+		return
+	}
+	columns := stmt.Cols
```

Comment:
This variable is used just once and its definition is very simple.
Just inline it.

Reply:
https://github.com/cloudspannerecosystem/harbourbridge/blob/99db369b2b826fc24ffbb760f3b3700cd0e04b0f/mysql/mysqldump.go#L170
Done.